### PR TITLE
Use a consistent pattern for exposing service dependencies

### LINKED ISSUE #20487: Make service dependencies protected virtual
~~And add an API consistency test~~ Split out into #25464 for MQ

### PR DESCRIPTION
**Pull request by ajcvickers**
Fixes #20487

### CODE REVIEW

File: src/EFCore.Relational/Scaffolding/ProviderCodeGenerator.cs [hunk 1]
```diff
@@ -14,14 +14,14 @@ public abstract class ProviderCodeGenerator : IProviderConfigurationCodeGenerato
...
+        /// <param name="relationalDependencies"> The dependencies. </param>
+        protected ProviderCodeGenerator(ProviderCodeGeneratorDependencies relationalDependencies)
+            => RelationalDependencies = Check.NotNull(relationalDependencies, nameof(relationalDependencies));
 
         /// <summary>
-        ///     Parameter object containing dependencies for this service.
+        ///     Relational provider-specific dependencies for this service.
         /// </summary>
-        protected virtual ProviderCodeGeneratorDependencies Dependencies { get; }
+        protected virtual ProviderCodeGeneratorDependencies RelationalDependencies { get; }
```
Comment (AndriySvyryd) [inline src/EFCore.Relational/Scaffolding/ProviderCodeGenerator.cs:24]: Generally we only name it `RelationalDependencies` if there's a conflicting `Dependencies` on base. I don't think the breaking change is worth it, especially for relational-only services

Comment (ajcvickers) [reply src/EFCore.Relational/Scaffolding/ProviderCodeGenerator.cs:24]: I'm fine with either way.

Comment (ajcvickers) [reply src/EFCore.Relational/Scaffolding/ProviderCodeGenerator.cs:24]: But do you want it to be consistent? That is, change `relationalDependencies` to `dependencies` unless there are two kinds of dependencies? Or just leave them a mix of `relationalDependencies` and `dependencis`?

Comment (AndriySvyryd) [reply src/EFCore.Relational/Scaffolding/ProviderCodeGenerator.cs:24]: We should change them to `dependencies`

Comment (ajcvickers) [reply src/EFCore.Relational/Scaffolding/ProviderCodeGenerator.cs:24]: @AndriySvyryd Updated.

**Review by AndriySvyryd — APPROVED** [review]

**Pull request merged**